### PR TITLE
fix: update OS listing fee

### PR DESCRIPTION
**Pull request by cbachmeier**
## Description

- Update OS listing fee to be a flat 2.5% for all collections as opposed to 0.5% for collections with no royalties or 0% for collections with royalties as mentioned in their OS Pro announcement
- Use the fee from the ListingMarket object when creating OS's consideration items

_JIRA ticket:_ https://uniswaplabs.atlassian.net/browse/NFT-1193
_Slack thread:_ https://uniswapteam.slack.com/archives/C03MGBD2SNN/p1683553053120529
_Relevant docs:_ https://support.opensea.io/hc/en-us/articles/14068991090067

## Screen capture
Updated fees
Screen Shot 2023-05-08 at 08 51 30
Screen Shot 2023-05-08 at 08 58 52
Successfully listing a 721 and an 1155
Screen Shot 2023-05-08 at 08 50 55
Screen Shot 2023-05-08 at 08 59 06

### QA (ie manual testing)

- [x] Successfully list a 721 and a 1155 nft to OS

### CODE REVIEW

File: src/nft/components/profile/list/utils.ts [hunk 1]
```diff
@@ -227,11 +227,6 @@ export const getRoyalty = (listingMarket: ListingMarket, asset: WalletAsset) =>
   return baseFee * 0.01
 }
 
-// OpenSea has a 0.5% fee for all assets that do not have a royalty set
-export const getMarketplaceFee = (listingMarket: ListingMarket, asset: WalletAsset) => {
```
Comment (cbachmeier) [inline src/nft/components/profile/list/utils.ts:231]: OS now just charges the flat 2.5% so this extra logic is not needed

File: src/nft/utils/listNfts.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ const getConsiderationItems = (
   openSeaFee?: ConsiderationInputItem
 } => {
   const creatorFeeBasisPoints = asset?.basisPoints ?? 0
-  const openSeaBasisPoints = !asset?.basisPoints ? 50 : 0
+  const openSeaBasisPoints = (ListingMarkets.find((market) => market.name === 'OpenSea')?.fee ?? 0) * 100
```
Comment (cbachmeier) [inline src/nft/utils/listNfts.ts:62]: Previously OS would require a 50 basis point fee for collections with no creator royalties and 0 basis points if it had royalties but now it's just the flat 250 basis points

File: src/nft/components/profile/list/MarketplaceRow.tsx [hunk 3]
```diff
@@ -155,11 +149,11 @@ export const MarketplaceRow = ({
 
   const fees = useMemo(() => {
     if (selectedMarkets.length === 1) {
-      return getRoyalty(selectedMarkets[0], asset) + getMarketplaceFee(selectedMarkets[0], asset)
+      return getRoyalty(selectedMarkets[0], asset) + selectedMarkets[0].fee
     } else {
       let max = 0
       for (const selectedMarket of selectedMarkets) {
-        const fee = getRoyalty(selectedMarket, asset) + getMarketplaceFee(selectedMarket, asset)
+        const fee = getRoyalty(selectedMarket, asset) + selectedMarket.fee
```
Comment (JackShort) [inline src/nft/components/profile/list/MarketplaceRow.tsx:156]: is there a reason we break this up for if there is only one marketplace - might it be easier just to put it all in the loop?

File: src/nft/utils/listNfts.ts (hunk 2 above)
Comment (JackShort) [inline src/nft/utils/listNfts.ts:62]: can this not just be `ListingMarkets[1].fee` then you won't need the `?`s

**Review by JackShort — APPROVED** [review]
small nits

File: src/nft/utils/listNfts.ts (hunk 2 above)
Comment (cbachmeier) [reply src/nft/utils/listNfts.ts:62]: this allows us to be able to modify the array without causing downstream effects

File: src/nft/components/profile/list/MarketplaceRow.tsx (hunk 3 above)
Comment (cbachmeier) [reply src/nft/components/profile/list/MarketplaceRow.tsx]: good callout, updated it

**Review by tinaszheng — APPROVED** [review]

File: src/nft/components/profile/list/MarketplaceRow.tsx [hunk 4]
```diff
@@ -154,17 +148,13 @@ export const MarketplaceRow = ({
   )
 
   const fees = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (cbachmeier) [reply src/nft/components/profile/list/MarketplaceRow.tsx:150]: It's actually the addition of two fees! The marketplace fee and the creator royalties.

**Pull request merged**